esphome/components/esp32_ble_client/ble_client_base.cpp

```diff
@@ -250,21 +254,46 @@ void BLEClientBase::gap_event_handler(esp_gap_ble_cb_event_t event, esp_ble_gap_
                format_hex(bd_addr, 6).c_str());
       if (!param->ble_security.auth_cmpl.success) {
         ESP_LOGE(TAG, "[%d] [%s] auth fail reason = 0x%x", this->connection_index_, this->address_str_.c_str(),
                  param->ble_security.auth_cmpl.fail_reason);
       } else {
-        ESP_LOGV(TAG, "[%d] [%s] auth success. address type = %d auth mode = %d", this->connection_index_,
+        ESP_LOGI(TAG, "[%d] [%s] auth success. address type = %d auth mode = %d", this->connection_index_,
                  this->address_str_.c_str(), param->ble_security.auth_cmpl.addr_type,
                  param->ble_security.auth_cmpl.auth_mode);
       }
       break;
-    // There are other events we'll want to implement at some point to support things like pass key
-    // https://github.com/espressif/esp-idf/blob/cba69dd088344ed9d26739f04736ae7a37541b3a/examples/bluetooth/bluedroid/ble/gatt_security_client/tutorial/Gatt_Security_Client_Example_Walkthrough.md
+    case ESP_GAP_BLE_PASSKEY_REQ_EVT:
+      // Call the following function to input the passkey which is displayed on the remote device
+      ESP_LOGD(TAG, "[%d] [%s] ESP_GAP_BLE_PASSKEY_REQ_EVT: Authenticating with passkey", this->connection_index_,
+               this->address_str_.c_str());
+      esp_ble_passkey_reply(this->remote_bda_, true, this->pin_code_);
+      break;
+    case ESP_GAP_BLE_PASSKEY_NOTIF_EVT:
+      // The app will receive this event when the IO has Output capability and the peer device IO has Input capability.
+      // Show the passkey number to the user to input it in the peer device.
+      ESP_LOGD(TAG, "[%d] [%s] ESP_GAP_BLE_PASSKEY_NOTIF_EVT: Passkey: %06d (0x%x)", this->connection_index_,
+               this->address_str_.c_str(), param->ble_security.key_notif.passkey,
+               param->ble_security.key_notif.passkey);
+      break;
+    case ESP_GAP_BLE_NC_REQ_EVT:
+      // The app will receive this evt when the IO has DisplayYesNO capability and the peer device IO also has
+      // DisplayYesNo capability. Show the passkey number to the user to confirm it with the number displayed by peer
+      // device.
+      ESP_LOGW(TAG, "[%d] [%s] ESP_GAP_BLE_NC_REQ_EVT: Passkey: %06d (0x%x) (Not implemented: esp_ble_confirm_reply)",
+               this->connection_index_, this->address_str_.c_str(), param->ble_security.key_notif.passkey,
+               param->ble_security.key_notif.passkey);
+      // esp_ble_confirm_reply(param->ble_security.ble_req.bd_addr, true);
+      break;
+    case ESP_GAP_BLE_OOB_REQ_EVT: {
+      ESP_LOGW(TAG, "[%d] [%s] ESP_GAP_BLE_OOB_REQ_EVT (Not implemented: esp_ble_oob_req_reply)",
+               this->connection_index_, this->address_str_.c_str());
+      // uint8_t tk[16] = {1}; // If you paired with OOB, both devices need to use the same tk
+      // esp_ble_oob_req_reply(param->ble_security.ble_req.bd_addr, tk, sizeof(tk));
```

Comment:
Probably good to note the reason for the commented code here too.

```suggestion
      // We probably want to something like this, however it has not yet been tested/verified.
      // uint8_t tk[16] = {1}; // If you paired with OOB, both devices need to use the same tk
      // esp_ble_oob_req_reply(param->ble_security.ble_req.bd_addr, tk, sizeof(tk));
```